README.md

```diff
@@ -2,24 +2,155 @@
+site/laboratory.
+1. Once you are done, click again on the pencil button: you should be asked a confirmation that you want to save your
+shapefile modifications.
+1. Copy the files with `.shq`, `.shx` and `.dbf` extensions to the `shapes` directory of the repository checkout.
+
+### Styles
+
+### Build the container image
 
-## Usage
+As for any Docker container image that you want to build from a Dockerfile, go into the directory
+containing the Dockerfile and run the following command:
 
 ```
 $ docker build . -t indico-maps
-$ docker run -p 8080:8080 -it indico-maps
 ```
 
-Then, just connect to https://yourhost:8080.
+''Note: `indico-maps` will be the tag of (name referring to) the image. You can use any
+name you prefer: nothing in the configuration refers to this name.`
+
+## Running the tile server
+
+### tileserver configuration
+
+The `tileserver` configuration file is located in `tileserver/config.json` in the repository.
+It contains 2 dictionaries which have normally one key each. The key must be the same in both
+dictionaries and will be your site name (used in the tile server URL configured in Indico).
+
+The 2 dictionaries are:
+* `data`: it contains an entry `mbtiles` which is the name of the file that contains the map
+tiles. It must match the name of the file produced by `tilemaker`.
+* `styles`: this dictionary has 2 entries:
+  * `styles`: must reflect you style configuration, i.e. the `config.json` path relative to the
+  `styles` directory.
+  * `tilejson/bounds`: an array of GPS coordinates of your site OSM map (processed by tilemaker)
+  for the south-west and the north-east corners with `latitude` the `longitude` for each one.
+
+### Starting the tileserver
+
+The typical command to run the tileserver container built previously is:
+
+```
+$  docker run -dit --restart unless-stopped -p 8080:8080 indico-maps
+```
+
+*Note: if you used an image tag different from `indico-maps` when you built the container,
+use it when starting the container.*
+
+The command above runs the container as a daemon (`-d`) and ensures it is restarted when Docker
+starts (for example after a reboot). If you want to see the tileserver output (logs), you must
+use the command `docker attach` with the container ID returned by the `run` command. Type CTRL+C
+to exit `docker attach`.
+
+### Testing the tile server
+
+To test your tile server, the easiest way is to use a browser and connect to `https://yourhost:8080`.
+
+### Configuring https access to the server
+
+Since your production Indico instance is running on HTTPS, you 
+need to use HTTPS for the tile server as well but `tileserver` doesn't support `https` natively. 
+To allow access the tileserver through https,
+you need to put a reverse proxy in front of it (like ). In the proxy configuration, you need
+to ensure that you pass `Host`, `X-Forward-Host` and `X-Forwarded-Scheme` headers to the tileserver.
+With , this is done by adding the following lines to the host configuration:
```

Comment:
"With ," - I guess there's something missing here? I would mention that this example is using `nginx`, but that you can use whatever you'd like.

Reply:
thanks for spotting it. `nginx` was missing!